Sources/Datadog/Core/Utils/Sampler.swift

```diff
@@ -0,0 +1,23 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+
+/// Sampler, deciding if events should be sent do Datadog or dropped.
+internal struct Sampler {
```

Comment:
/nit could be handy to make this `ExpressibleByFloatLiteral`, we would avoid having to update all tests.

```swift 
extension Sampler: ExpressibleByFloatLiteral {

    init(floatLiteral value: Float) {
        self.init(samplingRate: value)
    }
}
```

Reply:
Nice! I wish I could think of it before doing refactoring in the first place. But now as it is done, I think that having explicit `Sampler` type passed around just reads better - at least it's less magical.